packages/mdc-segmented-button/segmented-button/test/foundation.test.ts

```diff
@@ -0,0 +1,235 @@
+    mockAdapter.getSegments.and.returnValue([
+      {
+        'index': SELECTED,
+        'selected': true,
+        'segmentId': 'segment' + SELECTED
+      }, {
+        'index': UNSELECTED,
+        'selected': false,
+        'segmentId': 'segment' + UNSELECTED
+      }
+    ]);
+    return {foundation, mockAdapter};
+  }
+
+  /*** #isSingleSelect ***/
+
+  it('if single select, #isSingleSelect returns true', () => {
+    const {foundation} = setupSingleSelectTest();
+    expect(foundation.isSingleSelect()).toBeTruthy();
+  });
+
+  it('if multi select, #isSingleSelect returns false', () => {
+    const {foundation} = setupMultiSelectTest();
+    expect(foundation.isSingleSelect()).toBeFalsy();
+  });
+
+
+  /*** #isSegmentSelected ***/
+
+  it('if segment at index is selected, #isSegmentSelected returns true', () => {
+    const {foundation, mockAdapter} = setupSegmentTest();
+    let selectedSegment = mockAdapter.getSegments()[SELECTED];
+    expect(foundation.isSegmentSelected(selectedSegment.index)).toBeTruthy();
+  });
+
+  it('if segment with segmentId is selected, #isSegmentSelected returns true', () => {
+    const {foundation, mockAdapter} = setupSegmentTest();
+    let selectedSegment = mockAdapter.getSegments()[SELECTED];
+    expect(foundation.isSegmentSelected(selectedSegment.segmentId)).toBeTruthy();
+  });
+
+  it('if segment at index is not selected, #isSegmentSelected returns false', () => {
+    const {foundation, mockAdapter} = setupSegmentTest();
+    let unselectedSegment = mockAdapter.getSegments()[UNSELECTED];
+    expect(foundation.isSegmentSelected(unselectedSegment.index)).toBeFalsy();
+  });
+
+  it('if segment with segmentId is not selected, #isSegmentSelected returns false', () => {
+    const {foundation, mockAdapter} = setupSegmentTest();
+    let unselectedSegment = mockAdapter.getSegments()[UNSELECTED];
+    expect(foundation.isSegmentSelected(unselectedSegment.segmentId)).toBeFalsy();
+  });
+
+  it('if no segment at index, #isSegmentSelected returns false', () => {
+    const {foundation} = setupSegmentTest();
+    expect(foundation.isSegmentSelected(-1)).toBeFalsy();
+  });
+
+  it('if no segment with segmentId, #isSegmentSelected returns false', () => {
+    const {foundation} = setupSegmentTest();
+    expect(foundation.isSegmentSelected('-1')).toBeFalsy();
+  });
+
+  /*** #getSelectedSegments ***/
+
+  it('#getSelectedSegments returns selected segments', () => {
+    const {foundation, mockAdapter} = setupSegmentTest();
+    let selectedSegment = mockAdapter.getSegments()[SELECTED];
+    let selectedSegments = foundation.getSelectedSegments();
+    
+    expect(selectedSegments.length).toEqual(1);
+    expect(selectedSegments[0].index).toEqual(selectedSegment.index);
+    expect(selectedSegments[0].selected).toBeTruthy();
+    expect(selectedSegments[0].segmentId).toEqual(selectedSegment.segmentId);
+  });
+
+  /*** #selectSegment ***/
```

Comment:
Use `describe()` function to group tests. 

See data table's foundation test as an example.

Reply:
Thanks for the example!